### DISCUSSION TITLE
Support formatting for Levels

### DISCUSSION THREAD

**Discussion by chenna-reddy** [atoti/atoti#644] [General]
## Description

As of now, it is only possible to specify formatters for measures. It should be possible to specify formatted levels. Same formatters should also be used while displaying filters. 

## Context

I have a Date (No time) column in a table, which I would like to use for filtering. However, it always displayed as time (with time: 00:00). In financial applications we can't use this for specifying business date, value date etc....

Please support specifying formatters for Levels and use the same formatter in filtering panels too.

**Comment by tibdex** [comment]
> I have a Date (No time) column in a table, which I would like to use for filtering. However, it always displayed as time (with time: 00:00)

Can you check the data type of your level (is it a [LOCAL_DATE](https://docs.atoti.io/0.6.3/lib/atoti.type.html#atoti.type.LOCAL_DATE) or a [LOCAL_DATE_TIME](https://docs.atoti.io/0.6.3/lib/atoti.type.html#atoti.type.LOCAL_DATE_TIME))? `LOCAL_DATE` don't display `00:00:00` for the time as you can see [here](https://docs.atoti.io/0.6.3/lib/atoti.html#atoti.date_shift).

**Comment by chenna-reddy** [comment]
I'm using `tt.type.LOCAL_DATE`.  

> BTW, I don't define filters programmatically, it's defined using GUI. So I want to specify formatted at **level**.

#### My setup:
1. define Table definition using `session.create_table(...)`
2. Use `table.load_sql(query, url)` to load data from DB.

Thought it might be useful, as I have observed I can't really use `tt.type.NULLABLE_INT` or `tt.type.NULLABLE_INT` with this setup. So it might be an issue only with `atoti-sql` integration and types might be getting mixed up.
